File: Solutions/17. Letter Combinations of a Phone Number/17. Letter Combinations of a Phone Number.go
func letterCombinations(digits string) []string {
    var res []string
    if len(digits)==0{
        return res
    }
    numbers:=[]string{"0", "1", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"}
    res=append(res,"")
    for i:=0;i<len(digits);i+=1{
        x:=digits[i]-'0'
        for len(res[0])==i{
            t:=res[0];res=res[1:]
            for _,s:=range numbers[x]{
                res=append(res,t+string(s))
            }
        }
    }
    return res
}

File: Solutions/17. Letter Combinations of a Phone Number/17. Letter Combinations of a Phone Number.cpp
class Solution {
public:
    vector<string> letterCombinations(string digits) {
        vector<string> res;
        if(digits.size()==0){
            return res;
        }
        vector<string> numbers={"0", "1", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"};
        res.push_back("");
        for(int i=0;i<digits.size();++i){
            int x=digits[i]-'0';
            while(res[0].size()==i){
                string t=res[0];
                res.erase(res.begin());
                for(char s:numbers[x]){
                    res.push_back(t+s);
                }
            }
        }
        return res;
    }
};